phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -673,16 +668,21 @@ private boolean extractTablesAndIndexes(PhoenixConnection conn) {
         return indexInfos;
     }
 
+    @VisibleForTesting
+    public String getViewSql(String tableName, String schemaName) {
+       return "SELECT DISTINCT COLUMN_FAMILY, TENANT_ID FROM "
+                + "SYSTEM.CHILD_LINK "
+                + "WHERE TABLE_NAME = \'" + tableName + "\'"
+                + (StringUtil.EMPTY_STRING.equals(schemaName) ? " AND TABLE_SCHEM = \'"
```

Comment:
The order got messed up here (bug). Also, can you please add isNullOrEmpty() in StringUtil class and use it here? Your argument to avoid NPE is sort of incomplete since null string is being simply concatenated in the query string as if it was more valid empty schemaName - no difference functionally.

Reply:
It wasn't a complete push. I updated the commit. I think you are right about using isNullOrEmpty.